pysat/instruments/pysat_sgp4.py

```diff
@@ -0,0 +1,815 @@
+    inst.meta['sc_xhat_ecef_z'] = {'name':'sc_xhat_ecef_z',
+                                   'units':'',
+                                   'desc':'S/C attitude (x-direction, ram) unit vector, expressed in ECEF basis, z-component'}
+    
+    inst.meta['sc_zhat_ecef_x'] = {'name':'sc_zhat_ecef_x',
+                                   'units':'',
+                                   'desc':'S/C attitude (z-direction, generally nadir) unit vector, expressed in ECEF basis, x-component'}
+    inst.meta['sc_zhat_ecef_y'] = {'name':'sc_zhat_ecef_y',
+                                   'units':'',
+                                   'desc':'S/C attitude (z-direction, generally nadir) unit vector, expressed in ECEF basis, y-component'}
+    inst.meta['sc_zhat_ecef_z'] = {'name':'sc_zhat_ecef_z',
+                                   'units':'',
+                                   'desc':'S/C attitude (z-direction, generally nadir) unit vector, expressed in ECEF basis, z-component'}
+    
+    inst.meta['sc_yhat_ecef_x'] = {'name':'sc_yhat_ecef_x',
+                                   'units':'',
+                                   'desc':'S/C attitude (y-direction, generally south) unit vector, expressed in ECEF basis, x-component'}
+    inst.meta['sc_yhat_ecef_y'] = {'name':'sc_yhat_ecef_y',
+                                   'units':'',
+                                   'desc':'S/C attitude (y-direction, generally south) unit vector, expressed in ECEF basis, y-component'}
+    inst.meta['sc_yhat_ecef_z'] = {'name':'sc_yhat_ecef_z',
+                                   'units':'',
+                                   'desc':'S/C attitude (y-direction, generally south) unit vector, expressed in ECEF basis, z-component'}    
+    
+    # check what magnitudes we get
+    mag = np.sqrt(inst['sc_zhat_ecef_x']**2 + inst['sc_zhat_ecef_y']**2 + 
+                    inst['sc_zhat_ecef_z']**2)
+    idx, = np.where( (mag < .99999999999) | (mag > 1.00000000001))
+    if len(idx) > 0:
+        raise RuntimeError('Unit vector generation failure. Not sufficently orthogonal.')
+
+    return
+
+def calculate_ecef_velocity(inst):
+    """
+    Calculates spacecraft velocity in ECEF frame.
+    
+    Presumes that the spacecraft velocity in ECEF is in 
+    the input instrument object as position_ecef_*. Uses a symmetric
+    difference to calculate the velocity thus endpoints will be
+    set to NaN. Routine should be run using pysat data padding feature
+    to create valid end points.
+    
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        Instrument object
+        
+    Returns
+    -------
+    None
+        Modifies pysat.Instrument object in place to include ECEF velocity 
+        using naming scheme velocity_ecef_* (*=x,y,z)
+        
+    """
+    
+    x = inst['position_ecef_x']
+    vel_x = x.values[2:] - x.values[0:-2]
+    vel_x /= 2.
```

Comment:
Is there a reason not to do:

vel_x = (x.values[2:] - x.values[0:-2]) / 2.0

?

Reply:
Nope. Fixed.